BackEnd/routes/leaderboard.py

```
@@ -12,8 +12,16 @@ async def get_leaderboard(time_frame: str = "all", start_date: str = None, end_d
         date_filter = {}
-
-        if time_frame == "week":
+        
+        # Handle date filtering - if start_date/end_date provided, use them regardless of time_frame
+        if start_date and end_date:
+            try:
+                start = datetime.strptime(start_date, "%Y-%m-%d")
+                end = datetime.strptime(end_date, "%Y-%m-%d") + timedelta(days=1)
+                date_filter = {"contributions.timestamp": {"$gte": start, "$lte": end}}
+            except ValueError:
+                raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid date format. Use YYYY-MM-DD.")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)